src/Assignments/ProblemIframe.tsx

```diff
@@ -67,8 +69,35 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         // when problem changes, reset lastsubmitted and lastsaved
         setLastSubmittedAt?.(null);
         setLastSavedAt?.(null);
+        setLastSubmission({});
     }, [problem.id]);
 
+    const updateSubmitActive = _.throttle(() => {
+        const submitButtons = iframeRef.current?.contentWindow?.document.getElementsByName('submitAnswers') as NodeListOf<HTMLButtonElement>;
+        const problemForm = iframeRef.current?.contentWindow?.document.getElementById('problemMainForm') as HTMLFormElement;
+        // called only onLoad or after interaction with already loaded srcdoc - so form will exist, unless bad problemPath
+        // no console.error because exam problems (and static problems) will not have 'submitAnswers'
+        if (_.isNil(submitButtons) || _.isNil(problemForm)) {return;}
+
+        const isClean = _.isEqual(formDataToObject(new FormData(problemForm)), lastSubmission);
+
+        submitButtons.forEach((button: HTMLButtonElement) => {
+            if (isClean) {
+                button.setAttribute('disabled','true');
+                // invisibly stash the button's label (in case there are multiple submit buttons)
+                button.setAttribute('textContent', button.value);
+                button.setAttribute('value', 'Submitted');
+            } else {
+                button.removeAttribute('disabled');
+                if (button.textContent) {
+                    // put it back and clear the stash - just in case
+                    button.setAttribute('value', button.textContent);
```

Comment:
NIT: probably should be button.getAttribute

Reply:
getAttribute causes complaint with only the false-y check from the if statement.